src/main/java/pl/joegreen/sergeants/simulator/GameMap.java

```diff
@@ -0,0 +1,240 @@
+package pl.joegreen.sergeants.simulator;
+
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.joegreen.sergeants.api.response.GameUpdateApiResponse;
+import pl.joegreen.sergeants.api.response.ScoreApiResponse;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+class GameMap {
+    private final Logger LOGGER = LoggerFactory.getLogger(GameMap.class);
+
+    private ObjectMapper om = new ObjectMapper();
+
+    private final Tile[] tiles;
+    private int halfTurnCounter = 0;
+    private final int width;
+    private final int height;
+    private List<String> history = new ArrayList<>();
+
+
+    public GameMap(Tile[] tiles, int height, int width) {
+        if ((height * width) != tiles.length) {
+            throw new IllegalArgumentException("Incorrect height and width");
+        }
+        this.height = height;
+        this.width = width;
+        this.tiles = tiles;
+    }
+
+    public GameMap(int height, int width) {
+        this(new Tile[height * width], height, width);
+    }
+
+    public static GameMap of(final Replay replay) {
+        GameMap ret = new GameMap(replay.getMapHeight(), replay.getMapWidth());
+
+        Arrays.stream(replay.getMountains()).forEach(tileId -> ret.tiles[tileId] = new Mountain(tileId));
+
+        IntStream.range(0, replay.getCities().length).forEach(i -> {
+            int tileIndex = replay.getCities()[i];
+            int armySize = replay.getCityArmies()[i];
+            ret.tiles[tileIndex] = new City(tileIndex, armySize);
+        });
+
+        IntStream.range(0, replay.getGenerals().length).forEach(playerIndex -> {
+            int tileIndex = replay.getGenerals()[playerIndex];
+            String username = replay.getUsernames()[playerIndex];
```

Comment:
Sorry, should have been more verbose - I means just that one line, intellij says the "username" variable is not used anywhere :-)